app/models/content_item.rb

```diff
@@ -152,9 +164,16 @@ def includes_auth_bypass_id?(auth_bypass_id)
   end
 
   def register_routes(previous_item: nil)
-    return if self.schema_name.start_with?("placeholder")
-    return if previous_item && previous_item.route_set == self.route_set
-    self.route_set.register!
+    return unless should_register_routes?(previous_item: previous_item)
+
+    tries = Rails.application.config.register_router_retries
+    begin
+      route_set.register!
+    rescue GdsApi::BaseError
+      tries -= 1
+      retry if tries.positive?
+      raise
```

Comment:
Sorry - super minor, but might be nice for this to be `tries.positive? ? retry : raise`